broadinstitute/single_cell_portal_core
# Conceal "Show all" and "Hide all" for annotations with one group (SCP-3780)

This refines the custom scatter plot legend to only display "Show all" and "Hide all" links when multiple groups are available.

It also updates synthetic data to make that easily testable.

Before:
<img width="1675" alt="Before_single-label_legend_refinement__SCP_2021-12-16" src="https://user-images.githubusercontent.com/1334561/146435649-3ba5f9ba-1b26-4012-a4b5-840f7c22e3d5.png">

After:
<img width="1662" alt="After_single-label_legend_refinement__SCP_2021-12-16" src="https://user-images.githubusercontent.com/1334561/146435647-d95d8d0a-1353-4aaa-bc68-092c3c5e3362.png">

To test:
* Create / update "Tuberculosis in human female lymph" study: 
`echo "SyntheticStudyPopulator.populate('human_lymph_node')" | bundle exec rails console -e development`
* Navigate to its Study Overview
* Select clustering "cluster2.tsv" and annotation "has_one_label_in_cluster2"
* Verify that "Show all" and "Hide all" links are not shown
* Select clustering "cluster.tsv"
* Verify that "Show all" and "Hide all" links are shown

This satisfies SCP-3780.
